### ISSUE TITLE
kde plots fail when limits cut into posterior bulk

### ISSUE THREAD

**Issue by lukashergt** [williamjameshandley/anesthetic#105]
Changing limits in a way that cuts the posterior bulk messes with `kde` plots (see image below).

This was first encountered in #104, although unrelated to the fix therein. This might also be related to #95 and #99. 

### MWE:

```python
ns = NestedSamples(root="./tests/example_data/pc")
ns.limits['x0'] = (0, 5)
ns.plot_2d(['x0', 'x1', 'x2']);
```

![image](https://user-images.githubusercontent.com/18258042/85474717-f6edbf80-b5ac-11ea-9981-6b9fa6ff1279.png)

**Comment by lukashergt** [comment]
As mentioned in #99, a possible solution might be to separate `xmin` and `xmax` into compression/kde limits and plotting limits.

**Comment by williamjameshandley** [comment]
Another way would be to apply xmin and xmax limits, and then compress the remaining samples (rather than the opposite order as is currently done).

**Comment by lukashergt** [comment]
How would either case affect the behaviour at the boundary?

**Comment by lukashergt** [comment]
Fixed by #212 and #216. 

```python
ns = read_chains(root="./tests/example_data/pc")
fig, axes = ns.plot_2d(['x0', 'x1', 'x2']);
axes.iloc[0, 0].set_xlim(0, 1)
```
![image](https://user-images.githubusercontent.com/18258042/185777425-674ea9d6-337e-4d8a-9078-9222b8c8c2af.png)
